### ISSUE TITLE
Some messages are not read correctly by screen readers

### ISSUE THREAD

**Issue by Neurrone** [UnigramDev/Unigram#1235]
Hi,

I'm using the NVDA screen reader with Unigram and noticed some bugs with the way some messages are read because of missing information in the message's accessible label.

1. When a message is a reply to a previous one, the first line of the message being replied to should be part of the accessible label of the item to match what is being shown visually. For example, right now it reads something like "Bob. \r\nReplying to Alice. \r\nHi there.". I expect something like "Bob. \r\nReplying to Alice: first line of Alice's message... \r\nHi there."
2. There are some instances where the author of the message is not part of the accessible label of the message for replies. I've noticed this when the reply is a sticker. For example, "Replying to John Doe. \r\n👍 Sticker"

Fixing these bugs would greatly improve the experience for screen reader users, especially the first one so that it is clear what message is being replied to.

**Comment by Neurrone** [comment]
I've not done any c# / UWP before, but maybe fixing this might be a good way to get started. Will take a stab at fixing this over the weekend.

**Comment by FrayxRulez** [comment]
So this is definitely my fault: visually, if you have a group of messages sent by the same user, only the first one actually shows the sender name. I completely didn't consider the fact that if you're using the app with a screen reader this is completely unfriendly. I will add sender name to every message. Not sure about the reply tho.

**Comment by electronic-dk** [comment]
@FrayxRulez is it possible to add some meta info (like the sender's name) to the message for the screen reader to read but not actually render it?

**Comment by FrayxRulez** [comment]
> @FrayxRulez is it possible to add some meta info (like the sender's name) to the message for the screen reader to read but not actually render it?

Yes, this is what I will do!

**Comment by Neurrone** [comment]
@FrayxRulez thanks a lot. It would also be extremely helpful for messages that are replies if the first line of the message being replied to was also included in the accessible label, similar to how it is rendered visually.

**Comment by sukiletxe** [comment]
It would also be a good idea to add the localiced version of "you" to your own messages (like it is done outside of chats)

**Comment by sukiletxe** [comment]
Now only the "first line of the reply" thing (point 1) remains. Sadly, due to #1557 it will only solve the issue partially, as people won't only reply to text (myself included, of course).

**Comment by sukiletxe** [comment]
Replies thing is partially fixed now, as said in the previous comment.

**Comment by Neurrone** [comment]
The first line of the older message that is being replied to is still not read when reading a reply to it.

**Comment by sukiletxe** [comment]
Press tab, it should read the message being replied to. If B is a reply to A, pressing tab on B should read A.
